analyzer/tools/build-logger/src/ldlogger-util.c

```diff
@@ -86,15 +86,18 @@ int predictEscapedSize(const char* str_)
 
   while (*str_)
   {
-    if (strchr("\t\b\f\r\v\n ", *str_))
-      size += 2;
+    // 0x1B is '\e' (ESC)
+    if (strchr("\a0x1B\t\b\f\r\v\n ", *str_))
+      size += 3;
     else if (*str_ == '"' || *str_ == '\\')
       /* The quote (") and backslash (\) needs an extra escaped escape
          character because the JSON string literals are surrounded by quote
          by default. */
-      size += 3;
-
-    ++size;
+      size += 4;
+    else if ((unsigned char)*str_ < 0x20)
```

Comment:
I don't really like magic values, can we declare a variable to `0x20`? We use it multiple places.